### PR TITLE
[14.0][ENH] base_cancel_confirm, allow disable cancel confirm

### PR DESCRIPTION
**Pull request by kittiu**
By default, the cancel confirm will be disabled (to ensure no side effect on other module unit test)

To enable cancel confirm wizard, please add System Parameter (ir.config_parameter) for each extended module.

For example,

* sale_cancel_confirm, add `sale.order.cancel_confirm_disable = False`
* purchase_cancel_confirm, add `purchase.order.cancel_confirm_disable = False`

### CODE REVIEW

**Review by rousseldenis — COMMENTED** [review]
@kittiu I'm not sure of this as it should be disabled by default.

The cancel operation should flow normally except you enable the feature(that breaks that flow as waiting for the wizard confirmation)

**Comment by rousseldenis** [comment]
@kittiu

**Comment by kittiu** [comment]
@rousseldenis yes, it is disabled by default on the extended module such as sale_cancel_confirm https://github.com/OCA/sale-workflow/pull/2022
But in this base module, I just want to make it compatible with existing extended modules, not yet corrected as sale_cancel_confirm, to be enabled by defaul (no system param).

I try to make it backward compat.
WDYT?

File: base_cancel_confirm/model/base_cancel_confirm.py
Comment (rousseldenis) [inline base_cancel_confirm/model/base_cancel_confirm.py]: @kittiu What I wanted to say is this is the default if no ir config parameter entry.

Should be true here

Comment (kittiu) [reply base_cancel_confirm/model/base_cancel_confirm.py]: You are genius ! :)

Comment (rousseldenis) [reply base_cancel_confirm/model/base_cancel_confirm.py]: > You are genius ! :)

Don't over estimate me ;-)

**Review by rousseldenis — APPROVED** [review]
Code review

File: base_cancel_confirm/model/base_cancel_confirm.py
Comment (kittiu) [reply base_cancel_confirm/model/base_cancel_confirm.py]: I think you are rock star now. Very active contributor 🥇

File: base_cancel_confirm/model/base_cancel_confirm.py
Comment (ps-tubtim) [inline base_cancel_confirm/model/base_cancel_confirm.py]:
```suggestion
        res = self.env["ir.config_parameter"].sudo().get_param(key)
```
@kittiu Users other than admin can't work.
Selection_002

Comment (kittiu) [reply base_cancel_confirm/model/base_cancel_confirm.py]: OK thanks!

**Review by ps-tubtim — APPROVED** [review]

**Comment by kittiu** [comment]
@rousseldenis ok to merge?

**Comment by kittiu** [comment]
@rousseldenis can we merge this?

**Comment by rousseldenis** [comment]
> @rousseldenis can we merge this?

No superpowers here. @OCA/core-maintainers

File: base_cancel_confirm/model/base_cancel_confirm.py
Comment (simahawk) [inline base_cancel_confirm/model/base_cancel_confirm.py]: you can use `tools.str2bool` here

File: base_cancel_confirm/model/base_cancel_confirm.py
Comment (simahawk) [inline base_cancel_confirm/model/base_cancel_confirm.py:31]: I think `_enable` makes more sense, but it's likely a matter of taste :)

**Review by simahawk — COMMENTED** [review]
Can you please remove the odoo version from the commit?

File: base_cancel_confirm/model/base_cancel_confirm.py
Comment (kittiu) [reply base_cancel_confirm/model/base_cancel_confirm.py:31]: We use disable for backward compatibility too, there are extended modules what should be enable by default.

File: base_cancel_confirm/model/base_cancel_confirm.py
Comment (kittiu) [reply base_cancel_confirm/model/base_cancel_confirm.py:35]: Ok, thank you!

Comment (kittiu) [reply base_cancel_confirm/model/base_cancel_confirm.py]: Fixed, thank you!

**Review by Saran440 — APPROVED** [review]

**Comment by Saran440** [comment]
This PR is ready to merge. someone can merge, please?

**Review by simahawk — APPROVED** [review]
@kittiu LG, thanks.
One more thing, if you don't mind: please never include the odoo version in the commit -> when you port commits from one branch to another it makes little sense.
Also, if you keep the tag, improvements should be tagged w/ `IMP`.
Not a blocker of course :wink:

**Comment by Saran440** [comment]
@simahawk Thank you for your advise, I rename commit without version.

**Pull request merged**
